spec/models/rubric_criterion_spec.rb

```diff
@@ -2,4 +2,138 @@
+      expect(RubricCriterion.count).to be >0
+      criterion = RubricCriterion.first
+      criterion.max_mark = 0.5555555555
+      criterion.save
+      expect(criterion.max_mark).to eq 0.6
+    end
+
+    it 'find a mark for a specific rubric and result' do
+      assignment = create(:assignment)
+      grouping = create(:grouping, assignment: assignment)
+      submission = create(:submission, grouping: grouping)
+
+      complete_result = create(:complete_result, submission: submission)
+
+      rubric = create(:rubric_criterion, assignment: assignment)
+
+      mark = create(:mark, result: complete_result, markable: rubric)
+
+      expect(rubric.mark_for(complete_result.id))
+    end
+
+    it 'Set default levels' do
+      r = RubricCriterion.new
+      expect(r.set_default_levels)
+      r.save
+      expect(r.level_0_name).to eq(I18n.t('rubric_criteria.defaults.level_0'))
+      expect(r.level_1_name).to eq(I18n.t('rubric_criteria.defaults.level_1'))
+      expect(r.level_2_name).to eq(I18n.t('rubric_criteria.defaults.level_2'))
+      expect(r.level_3_name).to eq(I18n.t('rubric_criteria.defaults.level_3'))
+      expect(r.level_4_name).to eq(I18n.t('rubric_criteria.defaults.level_4'))
+    end
+
+    it 'be able to set all the level names at once' do
+      r = RubricCriterion.new
+      levels = []
+      0.upto(RubricCriterion::RUBRIC_LEVELS - 1) do |i|
+        levels << 'l' + i.to_s()
+      end
+      expect(r).to receive(:save).once
+      r.set_level_names(levels)
+      0.upto(RubricCriterion::RUBRIC_LEVELS - 1) do |i|
+        expect(r['level_' + i.to_s() + '_name']).to eq('l' + i.to_s())
+      end
+
+    end
+
+  end
+
+  context 'A rubric criterion assigning a TA' do
+    before(:each) do
+      @criterion = create(:rubric_criterion)
+      @ta = create(:ta)
+    end
+
+    it 'assign a TA by id' do
+      expect(@criterion.criterion_ta_associations.count).to be == 0, 'Got unexpected TA membership count'
+      @criterion.add_tas(@ta)
+      expect(@criterion.criterion_ta_associations.count).to be == 1, 'Got unexpected TA membership count'
+    end
+
+    it 'not assign the same TA multiple times' do
+      expect(@criterion.criterion_ta_associations.count).to be == 0, 'Got unexpected TA membership count'
+      @criterion.add_tas(@ta)
+      @ta.reload
+      @criterion.add_tas(@ta)
+      expect(@criterion.criterion_ta_associations.count).to be == 1, 'Got unexpected TA membership count'
+    end
+
+    it 'unassign a TA by id' do
+      expect(@criterion.criterion_ta_associations.count).to be == 0, 'Got unexpected TA membership count'
+      @criterion.add_tas(@ta)
+      expect(@criterion.criterion_ta_associations.count).to be == 1, 'Got unexpected TA membership count'
+      @ta.reload
+      @criterion.remove_tas(@ta)
+      expect(@criterion.criterion_ta_associations.count).to be == 0, 'Got unexpected TA membership count'
+    end
+
+    it 'assign multiple TAs' do
+      @ta1 = create(:ta)
```

Comment:
For local variables like `ta1`, don't use the `@` symbol (that's for instance attributes).